target_snowflake/db_sync.py

```diff
@@ -539,6 +560,107 @@ def load_csv(self, s3_key, count, size_bytes):
+            with connection.cursor() as cur:
+                # Create snowpipe if it does not exist
+                try:
+                    cur.execute(pipe_status_sql)
+                    self.logger.info("""snowpipe "{}" already exists!!""".format(pipe_name))
+                except ProgrammingError as excp:
+                    self.logger.info("""snowpipe "{}" does not exist. Creating...""".format(pipe_name))
+                    cur.execute(create_pipe_sql)
+                finally:
+                    cur.close()
+
+        # If you generated an encrypted private key, implement this method to return
+        # the passphrase for decrypting your private key.
+        # def get_private_key_passphrase(): #os.getenv('') #cartridge template grab
+        #     return ''
+
+        key_path = getattr(self.connection_config, "private_key_path", "/rsa_key.p8")
+        password = getattr(self.connection_config, "private_key_password", None)
+        with open(key_path, 'rb') as pem_in:
+            # pemlines = pem_in.read()
+            private_key_obj = load_pem_private_key(pem_in.read(),password=password,backend=default_backend())
+
+        private_key_text = private_key_obj.private_bytes(Encoding.PEM, PrivateFormat.PKCS8, NoEncryption()).decode('utf-8')
+        file_list=[s3_key]
+        self.logger.info(file_list)
+
+        ingest_manager = SimpleIngestManager(account=self.connection_config['account'].split('.')[0],
+                                        host=self.connection_config['account']+'.snowflakecomputing.com',
+                                        user=self.connection_config['user'],
+                                        pipe=pipe_name,
+                                        scheme='https',
+                                        port=443,
+                                        private_key=private_key_text)
+
+        # List of files, but wrapped into a class
+        staged_file_list = []
+        for file_name in file_list:
+            staged_file_list.append(StagedFile(file_name, None))
+
+        self.logger.info(staged_file_list)
+
+        try:
+            resp = ingest_manager.ingest_files(staged_file_list)
+        except HTTPError as e:
+            # HTTP error, may need to retry
+            self.logger.error(e)
+            exit(1)
+
+        # This means Snowflake has received file and will start loading
+        assert(resp['responseCode'] == 'SUCCESS')
+
+        # Needs to wait for a while to perform transfer, delete pipe after transfer
+        while True:
+            history_resp = ingest_manager.get_history()
+
+            if len(history_resp['files']) > 0:
+                self.logger.info('Ingest Report:\n')
+                self.logger.info(history_resp)
+                with self.open_connection() as connection:
+                    with connection.cursor() as cur:
+                        cur.execute(drop_pipe_sql)
+
+                cur.close()
```

Comment:
you don't/can't close this, the `with`context manager took care of that already.